libsolidity/codegen/YulUtilFunctions.h

```diff
@@ -314,7 +314,12 @@ class YulUtilFunctions
 
 	/// @returns the name of a function that does concatenation of variadic number of bytes
 	/// or fixed bytes
-	std::string bytesConcatFunction(std::vector<Type const*> const& _argumentTypes);
+	/// or Strings
+	std::string concatBytesOrStringFunctionCreate(
+		std::string functionName,
```

Comment:
I think you should derive the name from the function type kind inside the function